File: Data_Structures/C_C++/Linked_lists/reverseLinkedList.c
#include <stdio.h>

struct Node {
    int data;
    Node *next;
};

Node    *reverseLinkedList(struct Node *head)
{
    struct Node *prev = NULL;
    struct Node *next = NULL;

    while (head)
    {
        next = head->next;
        head->next = prev;
        prev = head;
        head = next;
    }
    return prev;
}

File: Data_Structures/javascript/Linked_lists/reverseLinkedList.js
const reverseLinkedList = head => {
    let prev = null;
    let next = null;
    while (head) {
        // Storing the pointer of the next index of the head
        next = head.next;

        // Moving the head's pointer to the previous ndoe
        head.next = prev;

        // changing the value of the prev to be the head pointer
        prev = head;

        // And then changing the value of the head to be the next one
        head = next;
    }
    return prev;
};
